Formula/deno.rb

```diff
@@ -1,76 +1,64 @@
 class Deno < Formula
-  desc "command-line JavaScript / TypeScript engine"
+  desc "Command-line JavaScript / TypeScript engine"
   homepage "https://deno.land/"
-  url "https://chromium.googlesource.com/chromium/tools/depot_tools.git",
-      :revision => "aec259ea62328ce39916607876956239fbce29b8"
-  version "0.3.3" # the version of the deno checkout, not a depot_tools version
-
-  # depot_tools/GN require Python 2.7+
-  # depends_on "python@2" => :build
+  url "https://github.com/denoland/deno.git",
+    :tag      => "v0.5.0",
+    :revision => "23aca36d52220efbfbc07abbe165bd2c9ade2009"
 
+  depends_on "llvm" => :build
+  depends_on "ninja" => :build
   depends_on "node" => :build
+  depends_on "python@2" => :build # GN require Python 2.7+
   depends_on "rust" => :build
 
   # https://bugs.chromium.org/p/chromium/issues/detail?id=620127
   depends_on :macos => :el_capitan
 
+  resource "gn" do
+    url "https://gn.googlesource.com/gn.git",
+      :revision => "64b846c96daeb3eaf08e26d8a84d8451c6cb712b"
+  end
+
   def install
-    # Add depot_tools in PATH
-    ENV.prepend_path "PATH", buildpath
-    # Prevent from updating depot_tools on every call
-    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
-    ENV["DEPOT_TOOLS_UPDATE"] = "0"
-    ENV["CHROMIUM_BUILDTOOLS_PATH"] = buildpath + "/deno/buildtools/"
+    # Build gn from source and add it to the PATH
+    (buildpath/"gn").install resource("gn")
+    cd "gn" do
+      system "python", "build/gen.py"
+      system "ninja", "-C", "out/", "gn"
+    end
+    ENV.prepend_path "PATH", buildpath/"gn/out"
 
-    # Initialize and sync gclient
-    system "gclient", "root"
-    system "gclient", "config", "--spec", <<~EOS
-      solutions = [
-        {
-          "url": "https://github.com/denoland/deno.git",
-          "managed": False,
-          "name": "deno",
-          "custom_deps": {},
-        },
-      ]
-      target_os = [ "mac" ]
-      target_os_only = True
-      cache_dir = "#{HOMEBREW_CACHE}/gclient_cache"
-    EOS
+    ENV["DENO_BUILD_MODE"] = "release"
+    inreplace "tools/third_party.py" do |s|
+      s.gsub! "gn_path = tp(depot_tools_path, \"gn\")", "gn_path = \"#{buildpath}/gn/out/gn\""
+      s.gsub! "ninja_path = tp(depot_tools_path, \"ninja\")", "ninja_path = \"#{Formula["ninja"].bin}/ninja\""
+    end
+    # restore node_modules symlink, which go converted to a copy of the folder by brew (is this a bug?)
+    rm_rf "node_modules"
+    ln_sf "third_party/node_modules", "node_modules"
```

Comment:
Yeah, this looks like a `brew` bug. Possible fix in https://github.com/chrmoritz/brew/commit/16a11b15bac53d0fea6b612c817c772443ab51c9. Thanks @hayd for discovering it.

Reply:
I've opened https://github.com/Homebrew/brew/pull/6141/files.

Reply:
Hmmm, I wonder why there's no jenkins CI build on your last commit.

Reply:
I think it is, but It's still pending because of the giant `gettext` job and the other PR queued.

Reply:
Does this mean these lines can go? (since your PR was merged) :)

Reply:
This would only work for users who are on `brew` master (`HOMEBREW_DEVELOPER` set or did run a dev command). Everyone else on the latest stable `brew` tag doesn't have this fix yet, so we have to wait until the next `brew` version is cut to finally remove this workaround.